Scons error: class with destructor marked 'final' cannot be inherited from

### Tested versions

- bad: e5b4ef8e9522e950033cbece39a31a4a76da19c1
- good: 4.2

### System information

Windows 11, Nvidia 3090, Vulkan forward+

### Issue description

```
=====
In file included from platform\windows\rendering_context_driver_vulkan_windows.cpp:35:
platform\windows/rendering_context_driver_vulkan_windows.h:55:50: error: class with destructor marked 'final' cannot be inherited from [-Werror,-Wfinal-dtor-non-final-class]
   41 | class RenderingContextDriverVulkanWindows : public RenderingContextDriverVulkan {
      |                                           final
   42 | private:
   43 |         const char *_get_platform_surface_extension() const override final;
   44 |
   45 | protected:
   46 |         SurfaceID surface_create(const void *p_platform_data) override final;
   47 |
   48 | public:
   49 |         struct WindowPlatformData {
   50 |                 HWND window;
   51 |                 HINSTANCE instance;
   52 |         };
   53 |
   54 |         RenderingContextDriverVulkanWindows();
   55 |         ~RenderingContextDriverVulkanWindows() override final;
      |                                                         ^
platform\windows/rendering_context_driver_vulkan_windows.h:41:7: note: mark 'RenderingContextDriverVulkanWindows' as 'final' to silence this warning
   41 | class RenderingContextDriverVulkanWindows : public RenderingContextDriverVulkan {
      |       ^
1 error generated.

=====
scons: *** [platform\windows\rendering_context_driver_vulkan_windows.windows.editor.double.x86_64.llvm.o] Error 1
=====
In file included from platform\windows\display_server_windows.cpp:43:
platform\windows/rendering_context_driver_vulkan_windows.h:55:50: error: class with destructor marked 'final' cannot be inherited from [-Werror,-Wfinal-dtor-non-final-class]
   41 | class RenderingContextDriverVulkanWindows : public RenderingContextDriverVulkan {
      |                                           final
   42 | private:
   43 |         const char *_get_platform_surface_extension() const override final;
   44 |
   45 | protected:
   46 |         SurfaceID surface_create(const void *p_platform_data) override final;
   47 |
   48 | public:
   49 |         struct WindowPlatformData {
   50 |                 HWND window;
   51 |                 HINSTANCE instance;
   52 |         };
   53 |
   54 |         RenderingContextDriverVulkanWindows();
   55 |         ~RenderingContextDriverVulkanWindows() override final;
      |                                                         ^
platform\windows/rendering_context_driver_vulkan_windows.h:41:7: note: mark 'RenderingContextDriverVulkanWindows' as 'final' to silence this warning
   41 | class RenderingContextDriverVulkanWindows : public RenderingContextDriverVulkan {
      |       ^
1 error generated.

=====
```

### Steps to reproduce

1. scons werror=yes use_llvm=yes use_mingw=yes precision=double compiledb=yes target=editor tests=yes debug_symbols=yes 
2. fails

### Minimal reproduction project (MRP)

None

--- a/platform/windows/rendering_context_driver_vulkan_windows.h
+++ b/platform/windows/rendering_context_driver_vulkan_windows.h
@@ -52,7 +52,7 @@ public:
 	};
 
 	RenderingContextDriverVulkanWindows();
-	~RenderingContextDriverVulkanWindows() override final;
+	~RenderingContextDriverVulkanWindows() override;
 };
 
 #endif // VULKAN_ENABLED

